packages/integration-sdk-runtime/src/logger/registerEventEmitters.ts

```diff
@@ -0,0 +1,117 @@
+import { IntegrationLogger } from '.';
+
+type EventType = 'multipleResolves'|'unhandledRejection'|'uncaughtException';
+
+type LifecycleErrorCallback = (err: Error, event: EventType) => void;
+
+// conforms to type MultipleResolveListener, see node_modules/@types/node/globals.d.ts
+type MultipleResolveListener = (type: 'resolve'|'reject', promise: Promise<any>, value: any) => void;
+
+function createMultipleResolveListener(callback: LifecycleErrorCallback): MultipleResolveListener {
+  return (type, promise, value) => {
+    if (type === 'reject') {
+      callback(value, 'multipleResolves');
+    }
+  };
+}
+
+// conforms to type UnhandledRejectionListener, see node_modules/@types/node/globals.d.ts
+type UnhandledRejectionListener = (reason: {} | null | undefined, promise: Promise<any>) => void;
+
+function createUnhandledRejectionListener(callback: LifecycleErrorCallback): UnhandledRejectionListener {
+  return (reason, promise) => {
+    callback(reason as Error, 'unhandledRejection');
+  };
+}
+
+// conforms to type UncaughtExceptionListener, see node_modules/@types/node/globals.d.ts
+type UncaughtExceptionListener = (error: Error) => void;
+
+function createUncaughtExceptionListener(callback: LifecycleErrorCallback): UncaughtExceptionListener {
+  return (error) => {
+    callback(error, 'uncaughtException');
+  };
+}
+
+interface RegisteredEventEmitters {
+  multipleResolveListener: MultipleResolveListener;
+  unhandledRejectionListener: UnhandledRejectionListener;
+  uncaughtExceptionListener: UncaughtExceptionListener;
+}
+
+/**
+ * Because individual integrations contain code in their `validateInvocation` and `integrationSteps`
+ * that could include unhandled rejections or multiple resolves, this function is exported as a convenience
+ * to catch and handle events emitted from an integration. This emitter should be instrumented as early
+ * as possible in the node process in order to handle any instrumentation exceptions.
+ *
+ * Before exiting the node process, unregister these event emitters using `unregisterEventEmitters`
+ * 
+ * @param callback
+ * @returns {RegisteredEventEmitters} Pass these listeners into `unregisterEventEmitters`
+ */
+export function registerEventEmitters(callback: LifecycleErrorCallback): RegisteredEventEmitters {
```

Comment:
I wonder if this is really registering error event handlers. 🤔

Reply:
Thanks for the review @aiwilliams! I agree with those naming conventions.